pkg/operation/botanist/component/extensions/operatingsystemconfig/downloader/downloader.go

```diff
@@ -197,6 +204,20 @@ WantedBy=multi-user.target`),
 		},
 	}
 
+	if gardenletfeatures.FeatureGate.Enabled(features.BootstrapTokenProvidedByWorker) {
```

Comment:
Actually, now having a second look, I'm wondering why we need this feature gate here in the first place. Can't we always create this file? You could adapt the executor script in the way that it does only consider the bootstrap token file if its content is not equal to the placeholder `<<BOOTSTRAP_TOKEN>>`. WDYT?

Reply:
As we want the worker bootstrap-token to be the default I agree with your proposal.
But: What if an os-extension version is used that does not know the `TransmitUnencoded` flag? I need to test whether it is just ignored, or whether this would lead to any problems (also: if an outdated os-extension accepts it then it would place it encoded on the VM).

Reply:
Good point, indeed. It seems the executor scripts need to be slightly "more intelligent", i.e., maybe it should check if the content of the file matches the regexp of a valid bootstrap token. Neither `<<BOOTSTRAP_TOKEN>>` nor `base64(<<BOOTSTRAP_TOKEN>>)` would pass.

Reply:
Looking again at this (always adding the file and removing the feature gate check there) is not solving any problems, or? It just pushes validation logic from here into the shell script. Why not leave it explicit in the gardener code. If the feature gate is `false` this file is never used - so we can also not add it. It is also clearer to the person reading the code that this file is just relevant if you enable the feature gate.

Reply:
IMO, this feature gate adds unnecessary complexity and avoidable cognitive overhead for operators during the deployment/configuration of gardenlet and the extensions. We are perfectly able to deliver this feature in a backwards-compatible way by enhancing the executor script, preventing mentioned downsides.
I guess the changes in the script would be minimal/just a few lines, so I'm not sure if any arguments w.r.t. code readability/maintainability apply/outweigh the mentioned complexity.